packages/wallet/src/redux/protocols/advance-channel/reducer.ts

```diff
@@ -315,3 +311,39 @@ function channelAdvanced(channel: ChannelState, commitmentType: CommitmentType):
     lastCommitment.commitmentType > commitmentType
   );
 }
+
+function nextCommitment(channel: ChannelState, commitmentType: CommitmentType): Commitment {
+  const lastCommitment = getLastCommitment(channel);
```

Comment:
Can't you just look at the round of commitments that is stored, count how many have the correct commitment type, and add one?

```
const commitments = getCommitments(channel); // fake method
const commitmentCount = commitments.find(c => c.commitmentType === commitmentType).length + 1;
```